core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -502,9 +608,16 @@ object ContainerPool {
   def props(factory: ActorRefFactory => ActorRef,
             poolConfig: ContainerPoolConfig,
             feed: ActorRef,
-            prewarmConfig: List[PrewarmingConfig] = List.empty) =
+            prewarmConfig: List[PrewarmingConfig] = List.empty)(implicit logging: Logging) =
     Props(new ContainerPool(factory, feed, prewarmConfig, poolConfig))
 }
 
 /** Contains settings needed to perform container prewarming. */
-case class PrewarmingConfig(count: Int, exec: CodeExec[_], memoryLimit: ByteSize)
+case class PrewarmingConfig(initialCount: Int,
+                            minCount: Int,
+                            maxCount: Int,
+                            exec: CodeExec[_],
+                            memoryLimit: ByteSize,
+                            ttl: Duration,
+                            threshold: Int,
+                            increment: Int)
```

Comment:
I think it may be good to separate the dynamic affects into a separate config, which is disabled by default so that operators need to opt-in to this new behavior. e.g. 
```
case class PrewarmingConfig(initialCount: Int,
                            exec: CodeExec[_],
                            memoryLimit: ByteSize,
                            reactiveConfig: Option[ReactivePrewarmingConfig]=None)

case class ReactivePrewarmingConfig(minCount: Int,
                            maxCount: Int,
                            ttl: Duration,
                            threshold: Int,
                            increment: Int)
```

Add some comments to indicate that the relationship of `threshold` and `increment` is; i.e. how do they determine the number of prewarms created and when.

Reply:
Great suggestion, updated accordingly.